### PR TITLE
Adds auto fill location input to explore page

### LINKED ISSUE #86: Add location auto complete to explore location search box
screen shot 2017-05-25 at 11 30 00 pm

- Located at `/explore`
- To use the `widgets/forms/location-input` component. The component has to be slightly modified to allow hiding of the 'Enter Address' button if and when needed.

### PR DESCRIPTION
**Pull request by geekyd**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds auto fill location input in explore page.

screenshot from 2017-05-26 16-05-04

Fixes #86 

@niranjan94 Please review.

### CODE REVIEW

File: app/components/widgets/forms/location-input.js [hunk 1]
```diff
@@ -24,7 +25,10 @@ export default Component.extend({
 
   actions: {
     showAddressView(show = true) {
-      this.set('addressViewIsShown', show);
+      let showAddress = this.get('showButton');
```
Comment (niranjan94) [inline app/components/widgets/forms/location-input.js]: Why is this required ? without the button this action will anyway not be triggered right ?

File: app/styles/components/explore/side-bar.scss [hunk 2]
```diff
@@ -24,3 +24,12 @@
   }
 }
 
+.location-input-container {
```
Comment (niranjan94) [inline app/styles/components/explore/side-bar.scss]: This whole css is not needed. You just have to remove the class `action` from `ui action input` when there is no button.

File: app/components/widgets/forms/location-input.js (hunk 1 above)
Comment (geekyd) [inline app/components/widgets/forms/location-input.js]: The action gets triggered when the user presses Return. So had to handle it.

Comment (niranjan94) [inline app/components/widgets/forms/location-input.js]: What is triggering the action ?

Comment (geekyd) [inline app/components/widgets/forms/location-input.js]: When the user pressed Return key the callback `placeChanged` is triggered which then triggers the `showAddressView` which hides the input field and shows the location in address form.

Comment (niranjan94) [inline app/components/widgets/forms/location-input.js]: @geekyd I completely forgot ... Use `widgets/forms/places-autocomplete` directly

File: app/styles/components/explore/side-bar.scss (hunk 2 above)
Comment (geekyd) [inline app/styles/components/explore/side-bar.scss]: I'm removing the CSS.

File: app/components/widgets/forms/location-input.js (hunk 1 above)
Comment (geekyd) [inline app/components/widgets/forms/location-input.js]: No problem, I will change it.

Comment (niranjan94) [inline app/components/widgets/forms/location-input.js]: Sorry about that ... Haven't been thinking straight lately.

**Comment by geekyd** [comment]
@niranjan94 I have made the changes. Please review. Thanks

File: app/templates/components/explore/side-bar.hbs [hunk 3]
```diff
@@ -3,7 +3,12 @@
 </div>
 <div class="item">
   <div class="ui input">
-    <input type="text" placeholder="enter location">
+    {{widgets/forms/places-autocomplete
+      id=inputId
```
Comment (niranjan94) [inline app/templates/components/explore/side-bar.hbs]: Remove this. Let ember add it's own id.

File: app/templates/components/explore/side-bar.hbs [hunk 4]
```diff
@@ -3,7 +3,12 @@
...
+      name=name
```
Comment (niranjan94) [inline app/templates/components/explore/side-bar.hbs]: Give it a name

File: app/templates/components/explore/side-bar.hbs [hunk 5]
```diff
@@ -3,7 +3,12 @@
...
+      value=placeName
+      type='text'
+      placeholder=placeholder}}
```
Comment (niranjan94) [inline app/templates/components/explore/side-bar.hbs]: Give it a placeholder

**Comment by geekyd** [comment]
Done
